modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java

```diff
@@ -221,9 +224,27 @@ private String _buildFragmentHTML(
 		return sb.toString();
 	}
 
-	private String _getAPIURL(ObjectEntry fdsEntryObjectEntry) {
-		Map<String, Object> properties = fdsEntryObjectEntry.getProperties();
+	private String _interpolateURL(String apiUrl, HttpServletRequest httpServletRequest){
+
+		ThemeDisplay themeDisplay =
+			(ThemeDisplay)httpServletRequest.getAttribute(
+				WebKeys.THEME_DISPLAY);
+
+		apiUrl = StringUtil.replace(apiUrl, "{siteId}", Long.toString(themeDisplay.getScopeGroupId()));
+		apiUrl = StringUtil.replace(apiUrl, "{scopeKey}", Long.toString(themeDisplay.getScopeGroupId()));
+		apiUrl = StringUtil.replace(apiUrl, "{userId}", Long.toString(themeDisplay.getUserId()));
 
+		if (StringUtil.contains(apiUrl, "{")) {
+			System.out.println("Unsupported parameter in API url: " + apiUrl);
```

Comment:
`System.out.println` is just for local debugging. For "real" logs, we need to use `LogFactoryUtil`, for example [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/java/com/liferay/frontend/data/set/sample/web/internal/portlet/FDSSamplePortlet.java#L186-L187).

In general, we don't often put logs in code, strong naming conventions go a log way to self-document code.

But, more importantly, I'm not sure if we want to add something like a warning or error line here. Yes, our own endpoint will result in an error, if it got to here there is a bug in previous DS manager code. But, in theory, a customer may have a legitimate "{" in their endpoint URL, and set it to DS persistence in their own way.

I'd just avoid complications and omit the log line here.

Reply:
That's interesting Marko, I asked Miguel to log this for debugging purposes, though I did not tell him about the LogFactoryUtil stuff, my fault :( 

My assumption was that there won't be any legitimate usage of `{parameterName}` in the URL for our APIs (probaby this is acceptable for general purpose URLs), meaning any such pattern needs substitution in all cases for headless urls. We can check this convention with Pablo just in case

Reply:
So, @markocikos @dsanz should I keep a log warning here?

Reply:
I suppose it's safe if you add a log line on warn level. Just make sure you wrap in check if warn scope is enabled. Something like [this](https://github.com/liferay/liferay-portal/blob/master/modules/apps/account/account-service/src/main/java/com/liferay/account/internal/retriever/AccountUserRetrieverImpl.java#L154-L156).

Reply:
Yep Pablo is ok with logging this also

Reply:
I've added it